### PR TITLE
Enable docker containers to communicate with remote roscores

### PR DESCRIPTION
**Pull request by Melanoneiro**
Docker containers need to have the ROS_MASTER_URI envvar set properly in order to be able to communicate with remove masters. This PR adds changes that create suppport for this.

### CODE REVIEW

**Comment by RaymondKirk** [comment]
Has this been tested @Melanoneiro? In both local and remote setups?

**Comment by Melanoneiro** [comment]
@nikostsagk and I have tested this on the robot remotely without issues, and I see no reason why running it with a roscore at localhost would not work, but I have asked him to do further testing and make sure there are no complications.

**Comment by nikostsagk** [comment]
I verify that we used this modification in both local and remote rosmaster scenarios and worked without implications :)

File: scripts/run_backend.sh [hunk 1]
```diff
@@ -71,7 +71,9 @@ fi
 if docker image inspect "$image_name" >/dev/null 2>&1 ; then
     echo "Image '${image_name}' exists locally"
     echo "Starting $image_name"
-    docker run --network host --gpus all --name $1_backend --rm -it $image_name
+    ROS_URI="${ROS_MASTER_URI:-"localhost:11311"}"
```
Comment (RaymondKirk) [inline scripts/run_backend.sh:74]: I would suggest we change this default value from "localhost:11311" to "http://localhost:11311" to conform with the actual ROS default string, then we can merge :-1:

Comment (Melanoneiro) [reply scripts/run_backend.sh]: Agreed and changed.

**Pull request merged**